src/Aggregation/DateHistogram.php

```diff
@@ -11,6 +11,55 @@ class DateHistogram extends Histogram
 {
     public const DEFAULT_TIMEZONE_VALUE = 'UTC';
 
+    private const CALENDAR_INTERVAL_VALUES = [
+        '1m',
+        'minute',
+        '1h',
+        'hour',
+        '1d',
+        'day',
+        '1w',
+        'week',
+        '1M',
+        'month',
+        '1q',
+        'quarter',
+        '1y',
+        'year',
+    ];
+
+    /**
+     * Set the interval by which documents will be bucketed.
+     *
+     * @param int|string $interval
+     *
+     * @return $this
+     */
+    public function setInterval($interval): self
+    {
+        if (\in_array($interval, self::CALENDAR_INTERVAL_VALUES, true)) {
+            return $this->setCalendarInterval($interval);
+        }
+
+        return $this->setFixedInterval($interval);
```

Comment:
We must cast the interval to string, as the two functions accepts a string only, not an integer

```suggestion
            return $this->setCalendarInterval((string) $interval);
        }

        return $this->setFixedInterval((string) $interval);
```

Reply:
This is not needed because
- We don't use `strict_types` so it's casted automatically
- No real `int` value are supported. It's always a number and a unit, like `2y`. 2 is not a supported value.

Reply:
but in the spirit to prepare to move to strict_types and PHP8, I see how
- adding casting to String
- and having the proper phpdoc
 
could help 3rd party users to properly migrate or prepare to migrate.
wdyt?

Reply:
I've trouble to understand.

If someone is calling
```
->setInterval(3);
```
the code already doesn't work.

Casting the value to string won't solve anything, since `'3'` is not a valid value either for DateHistogram interval.

IMHO, when creating CalendarIntervaDateHistograml or FixedIntervalDateHistogram, int support should be removed since it's not a valid value.

Anyway, I added a cast if it's a blocker for you.